tests/unit/type/list.tcl

```diff
@@ -1984,33 +1984,77 @@ foreach {pop} {BLPOP BLMPOP_RIGHT} {
     }
 }
 
-    test "List encoding conversion" {
-        set origin_conf [config_get_set list-max-listpack-size 3]
+    foreach {max_lp_size large} "3 $largevalue(listpack) -1 $largevalue(quicklist)" {
+        test "List listpack -> quicklist encoding conversion" {
+            set origin_conf [config_get_set list-max-listpack-size $max_lp_size]
+
+            # RPUSH
+            create_listpack lst "a b c"
+            r RPUSH lst $large
+            assert_encoding quicklist lst
+
+            # LINSERT
+            create_listpack lst "a b c"
+            r LINSERT lst after b $large
+            assert_encoding quicklist lst
+
+            # LSET
+            create_listpack lst "a b c"
+            r LSET lst 0 $large
+            assert_encoding quicklist lst
+
+            # LMOVE
+            create_quicklist lsrc "a b c $large"
+            create_listpack ldes "d e f"
+            r LMOVE lsrc ldes right right
```

Comment:
This failed with CROSSSLOT error in cluster mode (CI test-external-cluster).

Use a tag to make sure the the source and dest keys belong to the same slot, e.g.

            r LMOVE lsrc{t} ldes{t} right right

Reply:
Thx